src/Maestro/Client/src/MaestroApiCredential.cs

```diff
@@ -54,23 +57,78 @@ public override ValueTask<AccessToken> GetTokenAsync(TokenRequestContext _, Canc
         internal static MaestroApiCredential CreateUserCredential(string barApiBaseUri)
         {
             string appId = EntraAppIds[barApiBaseUri.TrimEnd('/')];
+            var requestContext = new TokenRequestContext(new string[] { $"api://{appId}/{USER_SCOPE}" });
 
-            // This is a usual credential obtained against an entra app through a browser sign-in
-            var credential = new InteractiveBrowserCredential(new InteractiveBrowserCredentialOptions
+            // This is a usual configuration for a credential obtained against an entra app through a browser sign-in
+            var credentialOptions = new InteractiveBrowserCredentialOptions
             {
                 TenantId = TENANT_ID,
                 ClientId = appId,
                 RedirectUri = new Uri("http://localhost"),
                 TokenCachePersistenceOptions = new TokenCachePersistenceOptions()
                 {
                     Name = "darc"
-                }
-            });
+                },
+            };
+
+            string authRecordPath = Path.Combine(MaestroApiOptions.AUTH_CACHE, $"{AUTH_RECORD_PREFIX}-{appId}");
+            var credential = GetCredentialFromCache(credentialOptions, requestContext, authRecordPath);
 
-            var requestContext = new TokenRequestContext(new string[] { $"api://{appId}/{USER_SCOPE}" });
             return new MaestroApiCredential(credential, requestContext);
         }
 
+        /// <summary>
+        /// Create interactive credential from an authentication record stored in local cache
+        /// Authentication record is a set of app and user-specific metadata used by the library to authenticate
+        /// </summary>
+        private static InteractiveBrowserCredential GetCredentialFromCache(
+            InteractiveBrowserCredentialOptions credentialOptions,
+            TokenRequestContext requestContext,
+            string authRecordPath)
+        {
+            InteractiveBrowserCredential credential;
+
+            if (!Directory.Exists(MaestroApiOptions.AUTH_CACHE))
+            {
+                Directory.CreateDirectory(MaestroApiOptions.AUTH_CACHE);
+            }
+
+            if (File.Exists(authRecordPath))
+            {
+                AuthenticationRecord authRecord;
```

Comment:
Do we need this variable? Maybe we can assign to `credentialOptions.AuthenticationRecord` directly?

Reply:
We can, will improve